lib/govspeak.rb

```diff
@@ -135,6 +135,33 @@ def insert_strong_inside_p(body, parser=Govspeak::Document)
       parser.new(body.strip).to_html.sub(/^<p>(.*)<\/p>$/,"<p><strong>\\1</strong></p>")
     end
 
+    extension('button', %r{
+      {button(.*?)} # match opening bracket and capture attributes
+        \s* # any whitespace between opening bracket and link
```

Comment:
I love this explanation of the regex, it makes it so clear

Reply:
I stole the idea from a faithful stack overflow explainer, makes me feel a little less guilty haha :P